pilot/pkg/networking/plugin/plugin.go

```diff
@@ -46,15 +50,21 @@ const (
 	Mixer = "mixer"
 )
 
-// ModelProtocolToListenerProtocol converts from a config.Instance to its corresponding plugin.ListenerProtocol
-func ModelProtocolToListenerProtocol(p protocol.Instance) ListenerProtocol {
+// ModelProtocolToListenerProtocol converts from a config.Protocol to its corresponding plugin.ListenerProtocol
+func ModelProtocolToListenerProtocol(node *model.Proxy, p protocol.Instance) ListenerProtocol {
 	switch p {
 	case protocol.HTTP, protocol.HTTP2, protocol.GRPC, protocol.GRPCWeb:
 		return ListenerProtocolHTTP
 	case protocol.TCP, protocol.HTTPS, protocol.TLS,
 		protocol.Mongo, protocol.Redis, protocol.MySQL:
 		return ListenerProtocolTCP
+	case protocol.UDP:
+		return ListenerProtocolUnknown
 	default:
+		if util.IsIstioVersionGE13(node) {
+			return ListenerProtocolAuto
```

Comment:
please add a feature flag here - I don't think we can safely turn on this feature by default a week before 1.3 freeze. This is an extremely risky change and has minimal testing.

Reply:
Before protocol sniffing is introduced, if listener protocol is unsupported, there will be no listener on that port. Now, we create a listener with protocol sniffing on that port. It's a safe change to me. I think we can add the guard in kube conversion if necessary as it's the only place that protocol `unsupported` is introduced. 

I think we can also wait until the istio/proxy broken is fixed. I did tons of e2e test locally. I also commit the test in this pr by changing some port name to random port name.

Reply:
As mentioned somewhere else, everywhere that you have the GE13 check should have the feature flag. New high risk features need to be protected by a feature flag so it can be disabled easily if it breaks something

It can be on by default if we really want, but must be able to turn off. Once it's stable in 1.4+ we can remove the flag

Reply:
I see. Do you think we should also disable protocol sniffing for test? e.g., e2e test

Reply:
Feature flag added.